Le Baluchon/Controller/ExchangeRates/ExchangeRatesViewController.swift

```diff
@@ -90,44 +95,41 @@ class ExchangeRatesViewController: UIViewController {
     
     // MARK: - Refresh
     @objc func refreshData() {
-        guard dataTask == nil else { return }
         refreshControl.beginRefreshing()
-        dataTask = networkClient.getData(completion: { (rates, error) in
-            self.dataTask = nil
-            self.refreshControl.endRefreshing()
+        viewModel.getRates()
+            .subscribe ({ event in
+                self.refreshControl.endRefreshing()
+                switch event {
+                case .success(let rates):
+                    self.converter = Converter(rates: rates.exchangeRates)
+                    self.converterView.fromCurrencyTF.text = "1"
+                    self.convert(amount: self.converterView.fromCurrencyTF.text)
+                    self.collectionView.reloadData()
 
-            guard let rates = rates as? ExchangeRates else {
-                let alertController = UIAlertController(title: "Erreur", message: error?.localizedDescription, preferredStyle: .alert)
-                alertController.addAction(UIAlertAction(title: "Ok", style: .default, handler: { (action) in
-                    self.dismiss(animated: true)
-                }))
-                self.present(alertController, animated: true)
-                return
-            }
-            
-            self.viewModel = ExchangeRatesViewModel(exchangeRates: rates)
-            self.converter = Converter(rates: rates.rates)
-            self.converterView.fromCurrencyTF.text = "1"
-            self.convert(amount: self.converterView.fromCurrencyTF.text)
-            self.collectionView.reloadData()
-        })
+                case .failure(let error):
+                    let alertController = UIAlertController(title: "Erreur", message: error.localizedDescription, preferredStyle: .alert)
+                    alertController.addAction(UIAlertAction(title: "Ok", style: .default, handler: { (action) in
+                        self.dismiss(animated: true)
+                    }))
+                    self.present(alertController, animated: true)
+                    
+                }
+            }).disposed(by: bag)
     }
 }
 
 // MARK: - CollectionView DataSource
 extension ExchangeRatesViewController: UICollectionViewDataSource {
     func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
-        guard let count = viewModel?.currencies.count else { return 0 }
-        
-        return count - 1
+        return viewModel.currencies.count == 0 ? 0 : viewModel.currencies.count - 1
```

Comment:
Pour info cette ligne de code est de la logique de VM. Elle aurai du etre de la forme: 
```suggestion
        return viewModel.getNumberOfCurrencies()
```
Comme ca tu la rend full testable de maniere unitaire :)